### PR TITLE
[ALLUXIO-2119] Remove use of whitebox in FileOutStreamTest

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2119

Use dependency injection in FileOutStream to remove the need for whitebox in the tests

### CODE REVIEW

File: core/client/src/main/java/alluxio/client/file/UnderFileSystemFileOutStream.java [hunk 1]
```diff
@@ -51,12 +51,40 @@
...
   /**
+   * Factory for creating an {@link UnderFileSystemFileOutStream}.
+   */
+  public static class Factory {
+    private static Factory sInstance;
+
+    /**
+     * @return an instance of a factory for creating {@link UnderFileSystemFileOutStream}
+     */
+    public static Factory get() {
```
Comment (calvinjia) [inline core/client/src/main/java/alluxio/client/file/UnderFileSystemFileOutStream.java]: Does this need to be synchronized?

Comment (aaudiber) [inline core/client/src/main/java/alluxio/client/file/UnderFileSystemFileOutStream.java]: Yes, good catch! I've updated to use `synchronized`

**Comment by aaudiber** [comment]
@calvinjia Comments addressed, PTAL

**Comment by calvinjia** [comment]
LGTM

**Comment by calvinjia** [comment]
@aaudiber could you resolve the merge conflicts?

**Pull request merged**